fetch-todolist/src/component/Home.js

```diff
@@ -1,76 +1,115 @@
-import { useState, useEffect } from "react";
-import AddToList from "./AddToDoList";
+import { useState, useEffect, useCallback } from "react";
 import ToDoList from "./ToDoList";
-import "./styling.css";
-import useFetch from "../customHook/useFetch";
 import DeleteFromList from "./DeleteFromList";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { getTodos, deleteTodo } from "../services/todos.services.js";
+import "./styling.css";
 
 const Home = () => {
-  const [list, setList] = useState([]);
   const [idToDel, setIdToDel] = useState();
-  const [isBlockHidden, setIsBlockHidden] = useState(false);
   const [isDelBlockHidden, setIsDelBlockHidden] = useState(false);
-  const { loading, error, data } = useFetch();
-  useEffect(() => {
-    setList(data);
-  }, [data]);
+  const pageSize = 4;
+  const [pageNumber, setPageNumber] = useState(1);
+  const [displayTodos, setTodos] = useState([]);
+  const [order, setOrder] = useState();
+  const queryClient = useQueryClient();
+  const [search, setSearch] = useState("");
+
+  const { data, isLoading, error } = useQuery(
+    ["todos", pageNumber, pageSize, order, search],
+    () => {
+      return getTodos({
+        _page: pageNumber,
+        _limit: pageSize,
+        _sort: "task",
+        _order: order,
+        task_like: search,
+      });
+    }
+  );
 
   const changeStatus = (check, id) => {
-    const updatedStatusArray = list.map((obj) => {
+    const updatedStatusArray = data.map((obj) => {
```

Comment:
Instad of map can we prefer to use find, As once are item is find then also it will keep iterating